File: app_escola_conectadaon/lib/models/comunicado.dart
import 'aluno.dart';
import 'professor.dart';

class Comunicado {
  int? id;
  String titulo;
  String mensagem;
  String? dataEnvio;
  Professor? professor;
  Aluno? aluno;
  Comunicado(
      {this.id,
      required this.titulo,
      required this.mensagem,
      this.dataEnvio,
      this.professor,
      this.aluno});
  Map<String, dynamic> toJson() => {
        'id': id,
        'titulo': titulo,
        'mensagem': mensagem,
        'professor': professor?.toJson(),
        'aluno': aluno?.toJson()
      };
  factory Comunicado.fromJson(Map<String, dynamic> json) => Comunicado(
      id: json['id'],
      titulo: json['titulo'] ?? '',
      mensagem: json['mensagem'] ?? '',
      dataEnvio: json['dataEnvio'],
      professor: json['professor'] != null
          ? Professor.fromJson(json['professor'])
          : null,
      aluno: json['aluno'] != null ? Aluno.fromJson(json['aluno']) : null);
}

File: api-escola-conectada/src/main/java/br/edu/escolaconectada/model/Comunicado.java
package br.edu.escolaconectada.model;
import jakarta.persistence.*; import java.time.LocalDateTime;
@Entity
public class Comunicado {
 @Id @GeneratedValue(strategy = GenerationType.IDENTITY) private Long id;
 private String titulo; @Column(length=2000) private String mensagem; private LocalDateTime dataEnvio = LocalDateTime.now();
 @ManyToOne private Professor professor; @ManyToOne private Aluno aluno;
 public Long getId(){return id;} public void setId(Long id){this.id=id;}
 public String getTitulo(){return titulo;} public void setTitulo(String titulo){this.titulo=titulo;}
 public String getMensagem(){return mensagem;} public void setMensagem(String mensagem){this.mensagem=mensagem;}
 public LocalDateTime getDataEnvio(){return dataEnvio;} public void setDataEnvio(LocalDateTime dataEnvio){this.dataEnvio=dataEnvio;}
 public Professor getProfessor(){return professor;} public void setProfessor(Professor professor){this.professor=professor;}
 public Aluno getAluno(){return aluno;} public void setAluno(Aluno aluno){this.aluno=aluno;}
}
